patcher/archiver.h

```diff
@@ -0,0 +1,43 @@
+#include <cstdint>
+#include <filesystem>
+#include <vector>
+
+#include "../archiver/archiver.h"
+
+extern const std::uint8_t data[]; // vector of compressed bytes --- stack of compressed files
+extern const uint8_t control[];   // vector of structs FileMemoryDefinition_t
+extern const uint8_t archive[];   // vector of FileArchiveDefinition_t
+
+namespace Unpacker
+{
+    class Unpacker
+    {
+
+    public:
+        /**
+         * @brief Structure describing uncompressed file and success of uncompress
+         *
+         */
+        struct UnCompressedDataDefinition
+        {
+            size_t OriginSize = 0;        ///< uncompressed size
+            std::unique_ptr<char[]> data; ///< uncompressed data
+            unsigned int crc32;           ///< crc32 value (after decompression)
```

Comment:
It unnecessarily to store both crc32 and IsCrcGood. Besides what are you suppose to do with broken data anyway?